src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryTable.cs

```diff
@@ -266,6 +266,10 @@ private static bool IsValueTypeErrorOrBlank<T>(FormulaValue val)
                     {
                         include = booleanValue.Value;
                     }
+                    else if (result is ErrorValue errorValue)
```

Comment:
This is the key change: if the filter predicate returns an error, then the whole row is a single error object.

Reply:
Correct. All the other changes are to support the concept of an "error record" in the test expected result.